src/WeakValueDict.jl

```diff
@@ -0,0 +1,222 @@
+# Weak value dictionaries
+#
+# This file is a modification of base/weakkeydict.jl from the Julia project.
+
+import Base: isempty, setindex!, getkey, length, iterate, empty, delete!, pop!, get, sizehint!, copy
+"""
+    WeakValueDict([itr])
+
+`WeakValueDict()` constructs a hash table where the values are weak
+references to objects which may be garbage collected even when
+used as keys in a hash table.
+"""
+mutable struct WeakValueDict{K,V} <: AbstractDict{K,V}
+    ht::Dict{K,WeakRef}
+    lock::ReentrantLock
+    finalizer::Function
+    dirty::Bool
+
+    # Constructors mirror Dict's
+    function WeakValueDict{K,V}() where V where K
+        t = new(Dict{K,Any}(), ReentrantLock(), identity, 0)
+        t.finalizer = v -> t.dirty = true
+        return t
+    end
+end
+function WeakValueDict{K,V}(kv) where V where K
+    h = WeakValueDict{K,V}()
```

Comment:
Perhaps call `sizehint!(h, XXX)` here with some suitable value XXX derived from kv? I don't know if there is a way to query a dictionary for its "size", though... Just a thought, feel free to ignore.

Reply:
`kv` has no length in general and it don't won't to write dozens of methods for different types.